k8sutils/pkg/cri/criwrapper.go

```diff
@@ -0,0 +1,142 @@
+package criwrapper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	criapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type CriClient struct {
+	conn   *grpc.ClientConn
+	client criapi.RuntimeServiceClient
+	Logger logr.Logger
+}
+
+// Default runtime endpoints
+var defaultRuntimeEndpoints = []string{
+	"unix:///run/containerd/containerd.sock",
+	"unix:///run/crio/crio.sock",
+	"unix:///var/run/cri-dockerd.sock",
+}
+
+// Connect attempts to establish a connection to a CRI runtime.
+func (rc *CriClient) Connect() error {
+	var err error
+
+	for _, endpoint := range defaultRuntimeEndpoints {
+		rc.Logger.Info("Attempting to connect to CRI runtime", "endpoint", endpoint)
+
+		rc.conn, err = grpc.NewClient(
+			endpoint,
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+		)
+
+		if err != nil {
+			continue
+		}
+
+		// Create a new RuntimeService client
+		rc.client = criapi.NewRuntimeServiceClient(rc.conn)
+		rc.Logger.Info("Successfully connected to CRI runtime", "endpoint", endpoint)
+		return nil
+	}
+
+	return fmt.Errorf("unable to connect to any CRI runtime endpoints: %v", defaultRuntimeEndpoints)
+}
+
+// GetContainerInfo retrieves the "info" field of the specified container.
+func (rc *CriClient) GetContainerInfo(ctx context.Context, containerID string) (map[string]string, error) {
+	containerID = extractContainerID(containerID)
+	if rc.client == nil {
+		return nil, fmt.Errorf("runtime client is not connected")
+	}
+
+	// Call the ContainerStatus API
+	response, err := rc.client.ContainerStatus(ctx, &criapi.ContainerStatusRequest{
```

Comment:
Maybe we want to put a timeout on that? In case the remote CRI does not respond in time / unavailable?

Reply:
done